pallets/collator-selection/src/lib.rs

```diff
@@ -540,7 +607,11 @@ pub mod pallet {
 			// `reward` is half of pot account minus ED, this should never fail.
 			let _success = T::Currency::transfer(&pot, &author, reward, KeepAlive);
 			debug_assert!(_success.is_ok());
-			<LastAuthoredBlock<T>>::insert(author, frame_system::Pallet::<T>::block_number());
+
+			// increment blocks this node authored
+			let mut authored_blocks = <BlocksPerCollatorThisSession<T>>::get(&author); // RAD: having this be an Option Query and throwing NotACandidate error would mess up testing code, hjow to resolve this?
```

Comment:
With a ValueQuery, this returns 0 if the validator doesnt exist, so it gets added.

Having this be an Option Query and throwing NotACandidate error would mess up testing code, how to resolve this?
(from genesis test code authors all blocks by origin 4, which is not part of the candidate or validator set at genesis)

Reply:
as validators are constant over a session and we're adding all validators to the map at the beginning of a new session, block production by a validator not in the map will not occur